src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -10953,8 +10953,14 @@ public override void VisitForEachIterationVariables(BoundForEachStatement node)
 
         public override BoundNode? VisitObjectInitializerMember(BoundObjectInitializerMember node)
         {
-            // Should be handled by VisitObjectCreationExpression.
-            throw ExceptionUtilities.Unreachable();
+            // We reach this point when we have just an identifier name in the initializer without an actual
+            // assignment, like "Value"
+            Debug.Assert(node.Syntax.Parent!.Kind() is SyntaxKind.ObjectInitializerExpression,
+                "Expected to visit this only in a member initializer with a missing assignment");
+
+            // We cannot perform any analysis here, as we have no real expression assignment or access
+            SetNotNullResult(node);
```

Comment:
Arguably that means we should set a `null` result, but I don't think I care that much.

Reply:
Most likely the user intends to assign a non-null value, so my choice was to assume the more likely scenario of assigning a non-null value after actually typing the `= expr` the user intends, thus avoiding the extra arising warnings in the code below, reducing friction while handling this erroneous case.

Reply:
Reverted this change as described in comments below